ARCs/arc-0027.md

```diff
@@ -0,0 +1,1687 @@
+  - **MUST** be a <a href="https://www.rfc-editor.org/rfc/rfc4122">UUIDv4</a> compliant string
+* `signer`:
+  - **MUST** be a base32 encoded public key with a 4 byte checksum appended as defined in <a href="https://developer.algorand.org/docs/get-details/accounts/#keys-and-addresses">keys and addresses</a>
+
+<sup>[Back to top ^][top]</sup>
+
+##### Sign Transactions Params
+
+```json
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "/schemas/sign-transactions-params",
+  "title": "Sign Transactions Params",
+  "description": "Sends a list of transactions to be signed by the provider(s)",
+  "type": "object",
+  "properties": {
+    "providerId": {
+      "type": "string",
+      "description": "A unique identifier for the provider",
+      "format": "uuid"
+    },
+    "txns": {
+      "type": "array",
+      "description": "A list of transactions to be signed by the provider(s)",
+      "items": {
+        "type": "object",
+        "properties": {
+          "authAddr": {
+            "type": "string",
+            "description": "The auth address if the sender has rekeyed"
+          },
+          "msig": {
+            "type": "object",
+            "description": "Extra metadata needed when sending multisig transactions",
+            "properties": {
+              "addrs": {
+                "type": "array",
+                "description": "A list of Algorand addresses representing possible signers for the multisig",
+                "items": {
+                  "type": "string"
+                }
+              },
+              "threshold": {
+                "type": "integer",
+                "description": "Multisig threshold value"
+              },
+              "version": {
+                "type": "integer",
+                "description": "Multisig version"
+              }
+            }
+          },
+          "signers": {
+            "type": "array",
+            "description": "A list of addresses to sign with",
+            "items": {
+              "type": "string"
+            }
+          },
+          "stxn": {
+            "type": "string",
+            "description": "The base64 encoded signed transaction"
+          },
+          "txn": {
+            "type": "string",
+            "description": "The base64 encoded unsigned transaction"
+          }
+        },
+        "required": ["txn"]
+      }
+    }
+  },
+  "required": [
+    "txns"
+  ]
+}
+```
+where:
+* `providerId`:
+  - **OPTIONAL** if omitted, all providers **MAY** respond
+  - **MUST** be a <a href="https://www.rfc-editor.org/rfc/rfc4122">UUIDv4</a> compliant string
+* `txns`:
+  - **MUST** have each item conform to the semantic of a transaction in [ARC-1](./arc-0001.md#semantic-of-wallettransaction)
+  - **MAY** be empty
+
+<sup>[Back to top ^][top]</sup>
+
+### Response Message Schema
+
+```json
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "/schemas/request-message",
```

Comment:
Small Nit: Looks like this one should be response-message

Reply:
Thank you! One can miss stuff like this in a copy + paste :sweat_smile:

Reply:
No worries! Trying out the specification in Kotlin/Java and was doing a lot of copy + paste myself!